File: getprime.rs
use std::env;


fn prime(num: u32) -> bool {
    let mut resp = true;    
    if num < 2 {
        resp = false;
    } else {
        if num > 2 {
            if num%2 == 0 {
                resp = false;
            } else {
                let mut i = 3;
                while i < num/2 {
                    if num%i == 0 {
                        resp = false;
                        break;
                    }
                    i += 2;
                }
            }
        }
    }
    resp
}


fn yieldprime(target: u32) -> u32 {
    let mut count = 1;
    let mut primes = 2;
    loop {
        if count == target {
            break;
        }
        primes += 1;
        if prime(primes) {
            count += 1;
        }
    }
    primes
}


fn main() {
    let args: Vec<String> = env::args().collect();
    let arg: u32 = args[1].trim().parse().expect("Please type a number!");
    println!("{}", yieldprime(arg));
}

File: getprime.hs
import System.Environment

prime :: Int -> Bool
_prime :: Int -> Int -> Bool
yieldprime :: Int -> Int
_yieldprime :: Int -> Int -> Int -> Int

prime num = _prime num 2

_prime num 2
    | num < 2 = False
    | num == 2 || num == 3 = True
    | mod num 2 == 0 = False
    | otherwise = _prime num 3

_prime num count
    | rest == 0 = False
    | div num 2 < count && rest /= 0 = True
    | otherwise = _prime num (count + 2)
    where rest = mod num count

yieldprime target = _yieldprime target 1 2

_yieldprime target count primes
    | count == target = primes
    | prime (primes + 1) == True = _yieldprime target (count + 1) (primes + 1)
    | otherwise = _yieldprime target count (primes + 1)

main = do
    num <- getArgs >>= readIO.head :: IO Int
    print (yieldprime num)
